server/application/application.go

```diff
@@ -1860,6 +1842,33 @@ func (s *Server) ListLinks(ctx context.Context, req *application.ListAppLinksReq
 	return finalList, nil
 }
 
+func (s *Server) getClusterObjectForDeepLinks(ctx context.Context, app *appv1.Application) (*unstructured.Unstructured, error) {
+	if err := argo.ValidateDestination(ctx, &app.Spec.Destination, s.db); err != nil {
+		log.WithFields(map[string]interface{}{
+			"application": app.GetName(),
+			"ns":          app.GetNamespace(),
+			"destination": app.Spec.Destination,
+		}).Warnf("cannot validate cluster, error=%v", err.Error())
+		return nil, nil
+	}
+	// check if user can access cluster
+	if err := s.enf.EnforceErr(ctx.Value("claims"), rbacpolicy.ResourceClusters, rbacpolicy.ActionGet, cluster.CreateClusterRBACObject(app.Spec.GetProject(), app.Spec.Destination.Server)); err != nil {
+		return nil, err
+	}
```

Comment:
I don't think we necessarily have to check for `clusters, get` access. I think it's fine to infer that access from their `applications, get`, since only admins can determine exactly what _parts_ of the cluster spec get exposed.

But I do think we should get the application's project and confirm that the destination cluster is an allowed destination before returning any info about that cluster.